components/custom/chat.tsx

```diff
@@ -0,0 +1,75 @@
+"use client";
+
+import { Attachment, Message } from "ai";
+import { useChat } from "ai/react";
+import { useState } from "react";
+
+import { Message as PreviewMessage } from "@/components/custom/message";
+import { useScrollToBottom } from "@/components/custom/use-scroll-to-bottom";
+
+import { MultimodalInput } from "./multimodal-input";
+import { Overview } from "./overview";
+
+export function Chat({
+  id,
+  initialMessages,
+}: {
+  id: string;
+  initialMessages: Array<Message>;
+}) {
+  const { messages, handleSubmit, input, setInput, append, isLoading, stop } =
+    useChat({
+      body: { id },
+      initialMessages,
+      maxSteps: 10,
+      onFinish: () => {
+        window.history.replaceState({}, "", `/chat/${id}`);
```

Comment:
any reasoun to use window instead of useRouter here?

Reply:
the intention was to do a shallow route push, and not reload the page